### ISSUE TITLE
[BUG] Easier gesture recognition

### ISSUE THREAD

**Issue by soulfoodz** [fermoya/SwiftUIPager#84]
**Is your feature request related to a problem? Please describe.**
I need to make a very exaggerated drag to scroll to the next item. I'd like to "flick" to the next page. TikTok would be the ideal scroll-ability/flick-ability example. Single page scrolling. 

**Describe the solution you'd like**
- Quicker drag recognition 
- Drag distance > 60pts  with velocity equals scroll to next
- Low velocity paired with the short drag distance to scroll
- Basically assume interaction means the user intends to scroll. Default to scrolling to next.

**Describe alternatives you've considered**
None yet. This is by far the best SwiftUI paging scroller I've found. Nice work. 

**Additional context**

**Comment by fermoya** [comment]
Hi @soulfoodz , so let me get this straight. What you’d need is a quicker animation to the the next page, meaning take less time to scroll (just like TikTok) and also make it easier to detect a scroll. Like I assume is kinda “hard” sometimes to scroll?

**Comment by tomhut** [comment]
Hey @fermoya Firstly thank you for creating this wonderful library, it is super nice.

This is something I have also run into. For my use case I have found the gesture needs to be very pronounced/exaggerated to increment to the next/previous page. It would be great if the library could use something like the gesture velocity or predictedEndLocation to help with this. I have put together a crude example below which uses fixed values rather than the item size, but it seems to generate the effect I am looking for.

```swift
 .onEnded({ (value) in
    let threshold: CGFloat = 200
    let newPage: Int
    if value.predictedEndLocation.x > threshold {
        newPage = self.currentPage - 1
    } else if value.predictedEndLocation.x < -threshold {
        newPage = self.currentPage + 1
    } else {
        newPage = currentPage
    }
    
    let animation = Animation.easeOut
    withAnimation(animation) {
        self.draggingOffset = 0
        self.pageIncrement = 1
        self.pageIndex = newPage
        self.draggingVelocity = 0
        self.lastDraggingValue = nil
    }
})
```

**Comment by fermoya** [comment]
@soulfoodz thanks a lot for the coffees, I appreciate it. Also both of you for the kind words.

@tomhut thanks for the code, I’ll give it a look in a bit. I do use velocity but a recent change I did might have bugged it. At the moment the user would need to drag half the screen or make a fast swipe (I think this second one might be “broken”). But for sure these values could be changed, I agree with you that they could be improved. 

I’ll give it a look as soon as I can (busy days today trying to find a new apartment), thank you both

**Comment by codetheweb** [comment]
I'm not sure if this is related or not, but it seems to require much more effort to swipe in 1.9.0 than 1.8.0. I'm downgrading back to 1.8.0 for now.

I can make a new issue for this if you want @fermoya, just wanted someone else to confirm first that my mind isn't playing tricks on me.

**Comment by fermoya** [comment]
@codetheweb no need for that. As I mentioned above, there was a fix in the last release that makes it more difficult, all right. I’ll work on that.

To be honest, it’s not really totally relate so you can open a new issue if you want. Up to you. I’ll be working on both together

**Comment by codetheweb** [comment]
Ok, sounds good - thank you.
No rush, 1.8.0 works perfectly for me. 😄

**Comment by fermoya** [comment]
@tomhut , @soulfoodz I've labeled this issue as a _bug_ because as @codetheweb mentioned, version 1.9.0 didn't have the dragging velocity updated. I've fixed this in [version 1.9.1](https://github.com/fermoya/SwiftUIPager/releases/tag/1.9.1).

Now, @tomhut and @soulfoodz . Could you please try this new release and let me know your thoughts? It should be easier now to scroll, but I'm not sure if this is what you wanted, @soulfoodz , or perhaps you'd be interested in a modifier to customize the animation?

**Comment by soulfoodz** [comment]
Playing with it now. It looks great! Thank you. A modifier would always be great, but as is, it's solid for my needs.

**Comment by fermoya** [comment]
@soulfoodz glad to hear that. I've added a new modifier `pagingAnimation` to allow customize the animation. I'll merge it into master and will release it in the next version.
